zerver/openapi/zulip.yaml

```diff
@@ -13931,8 +13931,9 @@ components:
             emoji's image can be found.
         still_url:
           type: string
+          nullable: true
           description: |
-            Only present when the emoji's image is animated.
+            Only non-null when the emoji's image is animated.
```

Comment:
This requires an API feature level bump if we change it.  (So does basically any other change in this file that isn't copyediting).  It's a necessary habit to allow the mobile apps to be implemented correctly with Zulip productions servers like Zulip Cloud that may run a version between major releases.

I also think we shouldn't make API changes like this just for mypy reasons -- one can use https://mypy.readthedocs.io/en/stable/more_types.html#mixing-required-and-non-required-items to make mypy happy.  But I suspect this is an improvement -- can you start an "#api design" conversation about this proposed API change and mention Greg to get quick feedback?  I suspect that the `nullable` API is better anyway, but it's worth using our API change process for this sort of decision.

Reply:
Also Mypy 0.930 now [supports](https://mypy-lang.blogspot.com/2021/12/mypy-0930-released.html) `typing_extensions.NotRequired` for individual `TypedDict` items ([PEP 655](https://www.python.org/dev/peps/pep-0655/)), which is a bit nicer than subclassing with `total=False` just to get some non-required items.

It is true, however, that non-required items weaken Mypy’s checking: Mypy won’t protect you from implicitly assuming the item is present when it might not be. So the API change still has potential value.

Reply:
I am not a fan of `NotRequired` fields, unless there are extreme performance requirements about what you send over the wire.  As Anders alludes to, it makes it difficult to know whether a codepath left off the value due to negligence or due to a proper business rule.

I just bumped the API version.  I am 99% sure that mobile does not try to use the field currently.  I haven't pushed yet, but this is the gist of my edit:

~~~
13946             **Changes**: New in Zulip 5.0 (added as optional field in
13947             feature level 97 and then made mandatory, but nullable, in
13948             feature level 113).